deploy-tool/main.go

```diff
@@ -132,26 +132,85 @@ func generateAndDeployKeyPair(
 	return p256Key, nil
 }
 
-func manifestExists(fqdn, dsp string) bool {
-	// Remove the locality name from the FQDN
-	path := fmt.Sprintf("https://%s/%s-manifest.json", fqdn, dsp)
+// globalManifestExists checks if a global data share processor manifest exists
+// relative to the manifest base URL provided, returning (true, nil) if so,
+// (false, nil) if not, and (false, error) if it could be determined
+func globalManifestExists(manifestBaseURL string) (bool, error) {
```

Comment:
Optimally this would be a thin wrapper for a method `fetchManifest` that could be called here and down in Fetcher so the status code and Sprintf-work could be consolidated by returning the response object in some way or other, but basically all this is a bit miserable so I think I agree with duplicating the code for this poor tool.

Reply:
I agree. Where I got hung up here is that in one case, all I care about is whether it exists, and in the other, I need the actual manifest. In any case this gets refactored away in #763 which is also why I was OK with letting this be ugly for that brief time between when this PR is merged and when that one lands.